csharp/src/Google.Protobuf/Reflection/EnumDescriptor.cs

```diff
@@ -128,12 +128,20 @@ public EnumValueDescriptor FindValueByName(string name)
         /// <summary>
         /// The (possibly empty) set of custom options for this enum.
         /// </summary>
-        [Obsolete("CustomOptions are obsolete. Use GetOption")]
+        [Obsolete("CustomOptions are obsolete. Use the Options property.")]
         public CustomOptions CustomOptions => new CustomOptions(Proto.Options?._extensions?.ValuesByNumber);
 
+        /// <summary>
+        /// The <c>EnumOptions</c>, defined in <c>descriptor.proto</c>.
+        /// Custom options can be retrieved as extensions of the returned message.
+        /// NOTE: A defensive copy is created each time this property is retrieved.
+        /// </summary>
+        public EnumOptions Options => (Proto.Options as IDeepCloneable<EnumOptions>)?.Clone();
```

Comment:
Can `Proto.Options` ever not be `IDeepCloneable<EnumOptions>`? Same question for all the other `as` casts. I prefer to do a regular cast unless there is a reason.

Reply:
Good point, removed the cast.